### PR TITLE
Python bindings for dnn module

### PR DESCRIPTION
**Pull request by ludv1x**
Python bindings for dnn module

Also merge of PR #738 is added

### CODE REVIEW

**Comment by ludv1x** [comment]
@AnnaPetrovicheva @Joker- 
I merged your changes and made some fixes.
At this moment all tests are passed and it seems that all examples works correctly -)

Further I will fix remaining warns and make some refactoring to make final merge into master.

**Comment by AnnaPetrovicheva** [comment]
@ludv1x Ok, great!

**Comment by alalek** [comment]
@ludv1x Could you please unpack and remove caffe.tar.gz, just commit files from this archive directly. To make it work you can take the first commit from #753.

**Comment by ludv1x** [comment]
@alalek Ok, I removed the archive and put compiled sources outside /src dir.

File: modules/dnn/cmake/OpenCVFindLibProtobuf.cmake [hunk 1]
```diff
@@ -23,19 +23,11 @@ else()
...
-    OUTPUT ${PROTOBUF_SRCS} ${PROTOBUF_HDRS}
-    COMMAND ${CMAKE_COMMAND} -E tar xzf ${CMAKE_CURRENT_SOURCE_DIR}/src/caffe/compiled/caffe.tar.gz
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    COMMENT "Unpacking compiled caffe protobuf files"
-    VERBATIM
-  )
+  set(PROTOBUF_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/protobuf/src ${CMAKE_CURRENT_SOURCE_DIR}/misc/caffe)
+  set(PROTOBUF_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/misc/caffe/caffe.pb.cc)
+  set(PROTOBUF_HDRS ${CMAKE_CURRENT_SOURCE_DIR}/misc/caffe/caffe.pb.h)
   set_source_files_properties(${PROTOBUF_SRCS} ${PROTOBUF_HDRS} PROPERTIES GENERATED TRUE)
```
Comment (alalek) [inline modules/dnn/cmake/OpenCVFindLibProtobuf.cmake]: Thanks you.

Could you please remove this line with "GENERATED TRUE" flag also.

Comment (ludv1x) [inline modules/dnn/cmake/OpenCVFindLibProtobuf.cmake]: Ok

**Comment by ludv1x** [comment]
I just now noticed two very strange errors on recently added configurations (Linux x64 Debug and Android pack).
The first one  [http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64_no_opt/builds/49/steps/test_face/logs/tests%20summary](url) tells that a test from face module is failed (but face module doesn't depend from dnn)
As for the second issue [http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_pack_android/builds/11/steps/Merge%20opencv_contrib%20with%20test%20branch/logs/stdio](url), it seems that test enviroment haven't acess to github.

So, Could we mege this PR, if problems with face module and android test server are resolved?

**Comment by alalek** [comment]
@ludv1x Please ignore there problems.

**Comment by vpisarev** [comment]
@ludv1x, thanks a lot for your excellent work! and sorry for delay :+1:

**Pull request merged**